File: c/buggy_code.c
#include <stdio.h>

// Missing semicolon
#define MAX 100

// Function missing return type
main() {
    printf("Starting buggy code\n");
    
    // Undeclared variable
    x = 10;
    
    // Missing semicolon
    y = 20
    
    // Wrong array syntax
    int arr[5];
    arr[5] = 10;  // Index out of bounds
    
    // Missing closing brace
    for (int i = 0; i < 5; i++ {
        printf("%d\n", i);
    
    // Wrong if syntax
    if (x = 10) {
        printf("x is 10\n");
    }
    
    // Missing parentheses
    if x > 5 {
        printf("x is greater than 5\n");
    }
    
    // Wrong string
    char *str = "Unclosed string;
    
    // Function call before declaration
    undefined_function();
    
    return 0;
}

// Missing semicolon after struct
struct Point {
    int x;
    int y;
}  // Missing semicolon

// Function with wrong syntax
void test(
    printf("Missing closing parenthesis\n");
}

File: javascript/buggy_code.js
// Buggy JavaScript code for testing

function main() {
    console.log("Starting buggy code")
    
    // Missing semicolons
    let x = 10
    let y = 20
    
    // Undeclared variable (in strict mode would be error)
    z = 30
    
    // Type error
    const num = 10;
    num.toUpperCase(); // Numbers don't have toUpperCase
    
    // Undefined variable
    console.log(undefinedVar);
    
    // Missing closing brace
    for (let i = 0; i < 5; i++ {
        console.log(i);
    
    // Wrong if syntax
    if x = 10 {
        console.log("x is 10");
    }
    
    // Missing parentheses
    if x > 5 {
        console.log("x is greater than 5");
    }
    
    // Unclosed string
    const str = "Unclosed string;
    
    // Wrong array access
    const arr = [1, 2, 3];
    console.log(arr[3]); // undefined, not error
    
    // Calling function before declaration
    myFunc();
    
    // Syntax error in function
    const test = () => {
        console.log("test")
        return // Missing value
    };
    
    // Invalid JSON
    const obj = {
        name: "John",
        age: 30,
        // Missing comma
        city: "New York"
        country: "USA"
    };
    
    // Using reserved word as variable
    const class = "test";
}

// Function missing closing brace
function myFunc() {
    console.log("This function has errors");
    // Missing closing brace
}

// Run the program
main(